addons/mrp/models/mrp_production.py

```diff
@@ -867,15 +867,26 @@ def write(self, vals):
                 if production.state == 'draft' and picking_type != production.picking_type_id:
                     production.name = picking_type.sequence_id.next_by_id()
 
-        res = super(MrpProduction, self).write(vals)
+        date_start_map = dict()
+        if 'date_start' in vals:
+            date_start_map = {
+                prod: vals['date_start'] - datetime.timedelta(days=prod.bom_id.produce_delay)
+                if prod.bom_id else vals['date_start']
+                for prod in self
+            }
+            res = True
+            for production in self:
+                res &= super(MrpProduction, production).write({**vals, 'date_start': date_start_map[production]})
+        else:
+            res = super().write(vals)
 
         for production in self:
-            if 'date_start' in vals and not self.env.context.get('force_date', False):
+            if production in date_start_map and not self.env.context.get('force_date', False):
```

Comment:
Regarding this part of the code:
```python
date_start_map = {
    prod: vals['date_start'] - datetime.timedelta(days=prod.bom_id.produce_delay)
    if prod.bom_id else vals['date_start']
    for prod in self
}
```
All records will be in `date_start_map` even if they have no BoM, thus I think keeping `if 'date_start' in vals` for the condition is enough (same for following conditions)
Let me know if I miss something